spec.bs

```diff
@@ -1505,6 +1513,46 @@ partial interface Navigator {
   </wpt>
 </div>
 
+<div algorithm>
+  The <dfn method for=Navigator>deprecatedURNtoURL(|urnOrConfig|, |send_reports|)</dfn>
+  method steps are:
+
+  1. Let |urn| be null.
+
+  1. If |urnOrConfig| is a {{USVString}}, set |urn| to |urnOrConfig|.
+
+  1. Otherwise, set |urn| to |urnOrConfig|'s [=fencedframeconfig/urn=].
+
+  1. If |urn| is <span class=XXX>TODO invalid</span>, [=exception/throw=] a {{TypeError}}.
+
+  1. Let |p| be [=a new promise=].
+
+  1. Let |global| be [=this=]'s [=relevant global object=].
+
+  1. Run the following steps [=in parallel=]:
+
+     1. Let |mapping| be |global|'s [=associated Document=]'s [=node navigable=]'s
+        [=navigable/traversable navigable=]'s [=traversable navigable/fenced frame config mapping=].
+
+     1. If |mapping|'s [=fenced frame config mapping/finalized config mapping=][|urn|], does not
+        [=map/exist=], [=queue a global task=] on the [=DOM manipulation task source=]
+        given |global|, to [=resolve=] |p| with {{undefined}}, and abort these steps.
+
+     1. Let |config| be |mapping|'s [=fenced frame config mapping/finalized config mapping=][|urn|].
+
+     1. [=Queue a global task=] on the [=DOM manipulation task source=] given |global|, to
+        [=resolve=] |p| with |config|'s [=fenced frame config/mapped url=]'s [=mapped url/value=].
+
+     1. If |send_reports| is true, run steps in |config|'s
+        [=fenced frame config/on navigate callback=].
```

Comment:
Why run these just on lookup? I assume that's what the imple does? It just feels a little strange.

Reply:
`deprecatedURNtoURL()` was added eons ago and the motivation was stated in https://github.com/WICG/turtledove/blob/main/Proposed_First_FLEDGE_OT_Details.md#advertisement-rendering as: "the `sendReports` parameter...is included to [promote accurate measurement](https://github.com/WICG/turtledove/blob/main/Proposed_First_FLEDGE_OT_Details.md#promoting-accurate-measurement-of-fledge-effectiveness.) and comparison of FLEDGE API utility by allowing auction winning URLs to be treated and used as they are in today’s ad auctions."
Basically we wanted a way to test Protected Audience (especially reporting scripts and APIs) when the URL coming from `deprecatedURNtoURL()` was rendered in a some other way that's not visible to the browser (so the browser doesn't know when to trigger Protected Audience reporting without the hint from `sendReports`).

Reply:
Any chance there's a WPT for this?

Reply:
It looks like we don't have WPTs for when `sendReports` is `true`, but I should note that `deprecatedURNtoURL()` is [going away at 3PCD](https://developers.google.com/privacy-sandbox/relevance/protected-audience-api/feature-status#:~:text=Note%3A%20The%20temporary%20navigator.deprecatedURNToURL()%20will%20be%20removed%20by%20third%2Dparty%20cookie%20deprecation.).